test/test_puma_server_partial_hijack.rb

```diff
@@ -0,0 +1,136 @@
+require_relative "helper"
+require "puma/events"
+require "puma/server"
+require "net/http"
+require "nio"
+require "ipaddr"
+
+class TestPumaServerPartialHijack < Minitest::Test
+  parallelize_me!
+
+  def setup
+    @host = "127.0.0.1"
+
+    @ios = []
+
+    @app = ->(env) { [200, {}, [env['rack.url_scheme']]] }
+
+    @log_writer = Puma::LogWriter.strings
+    @events = Puma::Events.new
+  end
+
+  def teardown
+    @server.stop(true)
+    assert_empty @log_writer.stdout.string
+    assert_empty @log_writer.stderr.string
+
+    # Errno::EBADF raised on macOS
+    @ios.each do |io|
+      begin
+        io.close if io.respond_to?(:close) && !io.closed?
+        File.unlink io.path if io.is_a? File
+      rescue Errno::EBADF
+      ensure
+        io = nil
+      end
+    end
+  end
+
+  def server_run(**options, &block)
+    options[:log_writer]  ||= @log_writer
+    options[:min_threads] ||= 1
+    @server = Puma::Server.new block || @app, @events, options
+    @port = (@server.add_tcp_listener @host, 0).addr[1]
+    @server.run
+    sleep 0.15 if Puma.jruby?
```

Comment:
What are we sleeping for here?

Reply:
See below comment.  Note that the above code came from test_puma_server.rb:
https://github.com/puma/puma/blob/8763c177a0bc8d7ec02b2faf1eda8efc687aa24b/test/test_puma_server.rb#L42-L49.

Since test_puma_server.rb currently has 1,565 lines, I thought creating another test file was a good idea.  Maybe it should be renamed and include 'full hijack' tests, thought we'd leave that for another commit.

Reply:
I just loaded JRuby head locally, and the statement `sleep 0.15 if Puma.jruby?` isn't needed.  It may be needed when CI is very busy.  I'll remove and we'll see what happens.  JFYI, it was added in Oct '22, so things may be different now.